myia_backend_python/myia_backend_python/python.py

```diff
@@ -0,0 +1,838 @@
+        if node.inputs:
+            fn = node.inputs[0]
+            if fn.is_constant(Primitive):
+                prim = fn.value
+                visit = getattr(self, f"_visit_{prim}", None)
+                if visit is None:
+                    return node.inputs[1:]
+                return visit(node)
+            else:
+                return node.inputs
+        elif node.is_constant_graph():
+            return [
+                fv
+                if not isinstance(fv, Graph)
+                else list(fv.manager.graph_constants[fv])[0]
+                for fv in node.value.free_variables_total
+            ]
+        return []
+
+
+def in_graph(g):
+    """Generate a filter callable for toposort. Copied from relay backend."""
+
+    def filter(node):
+        if node.graph is None:
+            return "follow"
+        elif node.graph is g:
+            return "follow"
+        else:
+            return "exclude"
+
+    return filter
+
+
+class _PythonConverter(Converter):
+    """Abstract class to convert values between myia and backend."""
+
+    def _default_convert(self, v, t):
+        """Default conversion. Just return given value."""
+        return v
+
+    def convert_array(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_scalar(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_nil(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_bool(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_universe(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_handle(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_tuple(self, v: tuple, t: AbstractTuple):
+        return self._default_convert(v, t)
+
+    def convert_tagged(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_type(self, v, t):
+        return self._default_convert(v, t)
+
+    def convert_random_state(self, v, t):
+        return self._default_convert(v, t)
+
+
+class PythonInputConverter(_PythonConverter):
+    """Convert an intermediate value to a backend value."""
+
+
+class PythonOutputConverter(_PythonConverter):
+    """Convert a backend value to an intermediate value."""
+
+
+class PythonConstantConverter(_PythonConverter):
+    """Convert constant values to printable values."""
+
+    def convert_scalar(self, v, t):
+        numpy_typename = type_to_np_dtype(t)
+        # For type names below, we return raw value.
+        if numpy_typename in ("bool", "int64", "uint64", "float64"):
+            return {
+                "bool": bool,
+                "int64": int,
+                "uint64": int,
+                "float64": float,
+            }[numpy_typename](v)
+        # Otherwise, we return a Python string code for this value.
+        return f"np.{numpy_typename}({v})"
+
+    def convert_dead(self, v, t):
+        # TODO How to convert dead value?
+        return None
```

Comment:
```suggestion
    def convert_dead(self, v, t):
        return "None"
```

A dead value is one that's not going to be used anywhere in the code, but could nonetheless not be removed from the graph because it would break some interface. It doesn't matter what it is, so returning None here is a good choice, but that should probably be a string.

Reply:
Ok !